### PR TITLE
Hands-On Assignment 4

### PR DESCRIPTION
**Pull request by hxshfx**

### CODE REVIEW

File: HandsOn/Group16/mappings/yarrrml_mapping.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  rdfs: "http://www.w3.org/2000/01/rdf-schema#"
+  sc: "http://www.schema.org#"
+
+mappings:
+    
+    TriplesMap1:
+        sources:
+            - [csv-updated.csv~csv, "$1"]
+            - [estaciones.csv~csv, "$2"]
+        s:
```
Comment (dachafra) [inline HandsOn/Group16/mappings/yarrrml_mapping.yml:12]: this should be
s: http://www.semanticweb.org/group16/ontologies/air-quality/$1(PUNTO_MUESTREO)

File: HandsOn/Group16/mappings/yarrrml_mapping.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+prefixes:
+  ns: "http://www.semanticweb.org/group16/ontologies/air-quality#"
+  rdfs: "http://www.w3.org/2000/01/rdf-schema#"
+  sc: "http://www.schema.org#"
+
+mappings:
+    
+    TriplesMap1:
+        sources:
+            - [csv-updated.csv~csv, "$1"]
```
Comment (dachafra) [inline HandsOn/Group16/mappings/yarrrml_mapping.yml:10]: add would encourage you to do a TriplesMap per csv!

File: HandsOn/Group16/mappings/yarrrml_mapping.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        s:
+            - http://www.semanticweb.org/group16/ontologies/air-quality/$1(PUNTO_MUESTREO)
+        po:
+            - [a, ns:Station]
+            - [ns:stationCode, $1(PUNTO_MUESTREO)]
+            - [rdfs:label, $2(NOMBRE)]
+            - [sc:address, $2(DIRECCION)]
+            - [sc:latitude, $2(LATITUD)]
+            - [sc:longitude, $2(LONGITUD)]
+            - p: ns:Station
```
Comment (dachafra) [inline HandsOn/Group16/mappings/yarrrml_mapping.yml:21]: here you have to set the Object Property for your ontology that relates stations with Mesurement

File: HandsOn/Group16/mappings/yarrrml_mapping.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                - mapping: TriplesMap2
+                  condition:
+                    function: equal
+                    parameters:
+                      - [str1, $1(PUNTO_MUESTREO)]
+                      - [str2, $2(ESTACION_ID)]
+    
+    TriplesMap2:
+        sources:
+            - [csv-updated.csv~csv, "$1"]
```
Comment (dachafra) [inline HandsOn/Group16/mappings/yarrrml_mapping.yml:32]: - [csv-updated.csv~csv] # remove the $1

File: HandsOn/Group16/mappings/yarrrml_mapping.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        s:
```
Comment (dachafra) [inline HandsOn/Group16/mappings/yarrrml_mapping.yml:33]: s: http://www.semanticweb.org/group16/ontologies/air-quality/$1(ID) (in one line)

**Review by dachafra — CHANGES_REQUESTED** [review]
take a look at my comments and provide a new version

**Comment by hxshfx** [comment]
We think this .yml file could be fixed, but the rmlmapper takes a **long** time to parse (at 20 minutes it didn't end). Tried with 10 rows in the csv works perfectly. The main problems are that we have so many rows in the csv (a little bit more than half a million) and that the identifier of the measurements is the concatenation of several columns on each row. Should we try to rename the measurement ID, or should we randomly remove some rows on the csv?

**Comment by dachafra** [comment]
Upload a CSV with 50-100 lines and the corresponding RDF generated, that would be enough.

**Review by dachafra — APPROVED** [review]

**Comment by dachafra** [comment]
well done!

**Pull request merged**